assets/prototype/domain/eventEditor/tickets/ticketPriceCalculator/hooks/useOnSubmitPrices.ts

```diff
@@ -1,68 +1,73 @@
 import { difference } from 'ramda';
+import { useCallback } from 'react';
 
 import { FnCallback, TpcFormData } from '../types';
 import { Price } from '../../../data/types';
 import { cloneAndNormalizePrice } from '../../../../shared/entities/prices/predicates/updatePredicates';
 import { useEntityMutator, EntityType } from '../../../../../application/services/apollo/mutations';
 import toBoolean from '../../../../../application/utilities/converters/toBoolean';
 import parsedAmount from '../../../../../application/utilities/money/parsedAmount';
+import { ModalSubmit } from '../../../../../application/ui/components/layout/eeditorModal';
 
 const useOnSubmitPrices = (existingPrices: Price[]): FnCallback => {
 	const { createEntity, updateEntity, deleteEntity } = useEntityMutator(EntityType.Price, null);
 	const { updateEntity: updateTicket } = useEntityMutator(EntityType.Ticket, null);
 	const existingPriceIds = existingPrices.map(({ id }) => id);
 
 	// Async to make sure that prices are handled before updating the ticket.
-	return async ({ ticket, prices = [] }: TpcFormData): Promise<void> => {
-		const updatedPriceIds = [];
-		const createdPriceIds = [];
+	return useCallback<ModalSubmit>(
+		async ({ ticket, prices = [] }: TpcFormData): Promise<void> => {
+			const updatedPriceIds = [];
+			const createdPriceIds = [];
 
-		// make sure to complete all price operations before updating the ticket
-		await Promise.all(
-			// covert the price operations into promises
-			prices.map((price: Price) => {
-				if (price.id === 'NEW_PRICE') {
-					return Promise.resolve(price);
-				}
-				const id = price.id;
-				const normalizedPriceFields = cloneAndNormalizePrice(price);
-				// if it's a newly added price
-				if (!id) {
+			// make sure to complete all price operations before updating the ticket
+			await Promise.all(
```

Comment:
Maybe: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of
?